### DISCUSSION TITLE
Fail validation if template is having Built-In Helpers / Custom Helpers / Partials

### DISCUSSION THREAD

**Discussion by ygangrades** [handlebars-lang/handlebars.js#1872] [Q&A]
Hi team,
Is it possible to fail the validation using handlebars functions if the template is having Built-In Helpers / Custom Helpers / Partials? 

I don't want the user to pass any conditional statements such as if-else for loop or helpers except Simple expressions, only simple expressions we wanted to allow. 

Is there any function or something which we can use to fail the validation if the user passes any Built-In Helpers / Custom Helpers / Partials?

We are using the java package of handlebars https://www.javadoc.io/doc/com.github.jknack/handlebars/latest/com/github/jknack/handlebars/Handlebars.html

**Comment by jaylinski** [comment]
The javascript package (this repo) provides ways to hook into the compiler/AST process:

https://github.com/handlebars-lang/handlebars.js/blob/v4.7.7/docs/compiler-api.md

For the java-package: you will have to ask here: https://github.com/jknack/handlebars.java
